### PR TITLE
Fixing a nil pointer in e2e

### PR DESCRIPTION
**Pull request by whaught**
## Proposed Changes

* I was paged for a nil pointer here.
    * Is there additional action we should take in this scenario?

### CODE REVIEW

**Comment by yegle** [comment]
The key clean up is best effort. We should be fine with keys left in db.

/lgtm

**Review by jeremyfaller — APPROVED** [review]

File: cmd/e2e-runner/main.go [hunk 1]
```diff
@@ -155,6 +155,7 @@ func realMain(ctx context.Context) error {
 		app, err := db.FindAuthorizedAppByAPIKey(deviceKey)
 		if err != nil {
 			logger.Errorf("device API key cleanup failed: %w", err)
+			return
```
Comment (chaodaiG) [inline cmd/e2e-runner/main.go:158]: Is this best effort? Should this be a panic instead?

Comment (whaught) [reply cmd/e2e-runner/main.go:158]: Yes. Best-effort according to @yegle. I'm not sure if we want to panic - if this were a concern we could consider an async db cleanup later perhaps

Comment (yegle) [reply cmd/e2e-runner/main.go:158]: I was wondering why this log looks like:

```device API key cleanup failed: %!w(*net.OpError=&{read tcp 0xc000371b90 0xc00040f2f0 0xc0006a3760})```

**Pull request merged**